src/components/templates/Poster/usePosterVideo.ts

```diff
@@ -0,0 +1,132 @@
+import { useEffect, useMemo, useState } from "react";
+import Video, { Room, LocalParticipant, RemoteParticipant } from "twilio-video";
+import firebase from "firebase/app";
+
+import { withId, WithId } from "utils/id";
+
+import { User } from "types/User";
+
+import { useUser } from "hooks/useUser";
+import { useWorldUsersById } from "hooks/users";
+
+// TODO: create useVideoRoom hook and divide this into two hooks & refactor it
+export const usePosterVideo = (venueId: string) => {
+  const [room, setRoom] = useState<Room>();
+  const [participants, setParticipants] = useState<
+    Array<LocalParticipant | RemoteParticipant>
+  >([]);
+  const [token, setToken] = useState<string>();
+  const [hasVideo, setHasVideo] = useState(false);
+
+  const { userId } = useUser();
+  const { worldUsersById } = useWorldUsersById();
+
+  const localParticipant = room?.localParticipant;
+
+  useEffect(() => {
+    (async () => {
+      if (!userId) return;
+
+      const getToken = firebase.functions().httpsCallable("video-getToken");
+      const response = await getToken({
+        identity: userId,
+        room: venueId,
+      });
+      setToken(response.data.token);
+    })();
+  }, [venueId, userId]);
+
+  useEffect(() => {
+    if (!localParticipant) return;
+
+    setParticipants((prevParticipants) => [
+      // Hopefully prevents duplicate users in the participant list
+      ...prevParticipants.filter(
+        (p) => p.identity !== localParticipant.identity
+      ),
+      localParticipant,
+    ]);
+  }, [localParticipant]);
+
+  useEffect(() => {
```

Comment:
This `useEffect` has no cleanup function, so every time it's called it will presumably setup another `Video.connect` connection and leave the old one running.

- https://reactjs.org/docs/hooks-effect.html#effects-with-cleanup

`src/components/templates/FireBarrel/useVideo.tsx` already handles this case by the looks of it.

Reply:
Irrelevant based on https://github.com/sparkletown/sparkle/pull/1299#discussion_r623416528